### ISSUE TITLE
Hangs on Incomplete but Finished stacks

### ISSUE THREAD

**Issue by jakewaggoner** [sjakthol/cfn-monitor#201]
There seems to be an issue when the stack fails to deploy successfully and also fails to cleanup correctly that causes the tool to hang. For instance, if I do a stack update and then it fails and tries to rollback and the rollback fails, then the stack is in the state `UPDATE_ROLLBACK_FAILED`. 

However, due to this https://github.com/sjakthol/cfn-monitor/blob/main/lib/cfn-events.js#L63 it will just sit there waiting for changes to occur. Same thing could be said for the state `DELETE_FAILED`. Anything that typically requires manual intervention will cause the tool to hang and not fail, thus we have to watch the output until we notice the failure.

**Comment by sjakthol** [comment]
Thanks for the report. I've made a fix and published it as version 2.4.2 to npm registry. cfn-monitor now also considers `FAILED` states as end states for stack operations and stops polling for more events when these occur. It should no longer hang but you still need to do some output parsing to figure out if the operation failed or not.

**Comment by jakewaggoner** [comment]
Wow thanks for the fast response. One thing to note is that I believe having it just check for FAILED will result in a premature exit when something like `UPDATE_FAILED` or `CREATE_FAILED` is shown.

**Comment by jakewaggoner** [comment]
> Wow thanks for the fast response. One thing to note is that I believe having it just check for FAILED will result in a premature exit when something like `UPDATE_FAILED` or `CREATE_FAILED` is shown.

My mistake. Since the code checks if the resource is of type `AWS::CloudFormation::Stack`, it will not stop until that is marked as failed. Thanks for the fix!

**Comment by sjakthol** [comment]
Yep, you are right. Only events from the stack is considered, and stack goes to FAILED or COMPLETED state only when the stack operation is actually over.

However, thanks to your comment I took another look at the code, did some extra tests and noticed that the earlier fix was buggy. This bug caused cfn-monitor to stop monitoring the stack when it moved to e.g. `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS` state. I've now fixed this issue and published version 2.4.3 to npm registry.
